### PR TITLE
Added dead mans switch contracts and base58 lib.

### PR DESCRIPTION
**Pull request by msinkec**

### CODE REVIEW

File: contracts/deadMansSwitchNLocktime.scrypt [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import "util.scrypt";
+
+/**
+ * Dead man's switch / last will contract.
```
Comment (xhliu) [inline contracts/deadMansSwitchNLocktime.scrypt:4]: Add comments explaining the difference btw the two versions of deadman

File: contracts/deadMansSwitchStatefull.scrypt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  
+    @state
+    PubKey pubKeyInheritor;
+
+    @state
+    int inheritanceSats;
+
+    // The point of the refresh function is just to replace the old UTXO 
+    // with an exact copy in a newer block.
+    public function refresh(SigHashPreimage txPreimage, Sig sig) {
```
Comment (xhliu) [inline contracts/deadMansSwitchStatefull.scrypt:27]: Add comments for each non-trivial property & function.

**Review by xhliu — APPROVED** [review]
🥇 looks good.


Add some comments before merging.

**Comment by msinkec** [comment]
OK, I added some comments and as we discussed, I also removed the nLocktime version of dead mans switch, because we came to the conclusion, that the same goal can be achieved ussing a simple P2PKH using nLocktime.

**Pull request merged**